scripts/rvnbag_play.py

```diff
@@ -271,37 +229,36 @@ def __init__( self, bagfile, loop):
         rospy.loginfo("...Done")
 
         # create clock publisher
-        self._clock_publisher = rospy.Publisher("/clock", Clock, queue_size=100)
+        self._clock_publisher = rospy.Publisher("/clock", Clock, queue_size=1, tcp_nodelay=True)
 
 
-    def _create_publisher( self, msg_cont ):
+    def _create_publisher( self, topic, msg_class ):
         try:
             new_pub = rospy.Publisher(
-                msg_cont.topic,
-                msg_cont.msg_class,
-                latch      = self.bag_reader.is_latched(msg_cont.topic),
-                queue_size = 1000 ) # RVN::FIX: queue_size=???
-
+                topic,
+                msg_class,
+                latch      = self.bag_reader.is_latched(topic),
+                queue_size = 16,
+                tcp_nodelay = True,
+            )
         except Exception as e:
-            self._set_terminate( ExitStatus.PUB_FAIL, "Cannot create publisher for '{}' topic: {}".format(msg_cont.topic,e))
+            self._set_terminate( ExitStatus.PUB_FAIL, "Cannot create publisher for '{}' topic: {}".format(topic,e))
+            return
 
-        self._all_publishers[msg_cont.topic] = new_pub
+        self._all_publishers[topic] = new_pub
 
 
-    def _publish_msg( self, msg_cont ):
+    def _publish_msg( self, msg ):
 
         # clock message is special and we handle it specially!
-        if msg_cont.topic == "/clock": return
-
-        if msg_cont.topic not in self._all_publishers:
-            self._create_publisher(msg_cont)
+        if msg["topic"] == "/clock": return
 
-        try:
-            msg = msg_cont.deserialize()
-        except Exception as e:
-            self._set_terminate( ExitStatus.BAD_DESERIAL, "Unable to publish msg to '{}' topic, cannot deserialize: {}".format(msg_cont.topic,e))
+        if msg["topic"] not in self._all_publishers:
+            self._create_publisher(msg["topic"],msg["raw_msg"][-1])
 
-        self._all_publishers[msg_cont.topic].publish(msg)
+        pmsg = rospy.msg.AnyMsg()
```

Comment:
Want to call out this bit for future reference, as it took me awhile to figure out how to use the `AnyMsg` class.

It allows you to read and publish a message without de-serializing it, or even having access to it's class definition. This is probably the single biggest source of CPU saving in this PR, as `rvnbag_play.py` was previously deserializing every message (some messages can be > 100mb) before publishing.